zookeeper-server/src/main/java/org/apache/zookeeper/server/ServerCnxnFactory.java

```diff
@@ -54,6 +54,7 @@
      */
     static final ByteBuffer closeConn = ByteBuffer.allocate(0);
 
+    private static String loginUser = System.getProperty("user.name", "<NA>");
```

Comment:
final?

Reply:
I think loginUser can not be marked as final